jetty-integrations/jetty-nosql/src/main/java/org/eclipse/jetty/nosql/mongodb/MongoSessionDataStore.java

```diff
@@ -315,12 +317,13 @@ public boolean delete(String id) throws Exception
     @Override
     public boolean doExists(String id) throws Exception
     {
-        DBObject fields = new BasicDBObject();
-        fields.put(__EXPIRY, 1);
-        fields.put(__VALID, 1);
-        fields.put(getContextSubfield(__VERSION), 1);
+//        DBObject fields = new BasicDBObject();
+//        fields.put(__EXPIRY, 1);
+//        fields.put(__VALID, 1);
+//        fields.put(getContextSubfield(__VERSION), 1);
 
-        DBObject sessionDocument = _dbSessions.findOne(new BasicDBObject(__ID, id), fields);
+        Bson filterId = Filters.eq(__ID, id);
+        Document sessionDocument = _dbSessions.find(filterId).first();
```

Comment:
I think the old code was trying to be efficient about the fields that we returned, ie only returning the fields that would be examined, rather than the whole set of serialized session attributes (which won't be examined). Does the new API provide a similar efficiency?